### PR TITLE
Use system_role only for x86_64 arch 

### PR DESCRIPTION
**Pull request by dzedro**
https://bugzilla.suse.com/show_bug.cgi?id=976558
https://build.suse.de/request/show/114291

### CODE REVIEW

File: tests/installation/system_role.pm [hunk 1]
```diff
@@ -12,9 +12,8 @@ use base "y2logsstep";
 use testapi;
 
 sub run() {
-    if (check_screen 'system-role-default-system') {
-        send_key 'alt-n';    # next
-    }
+    assert_screen 'system-role-default-system';
```
Comment (mimi1vx) [inline tests/installation/system_role.pm]: you are sure about this ? system-role is only in SLE12SP2+ ( and new TW )

File: products/sle/main.pm [hunk 2]
```diff
@@ -478,7 +478,9 @@ sub load_inst_tests() {
     loadtest "installation/addon_products_sle.pm";
     if (noupdatestep_is_applicable) {
         loadtest "installation/system_role.pm";
-        loadtest "installation/partitioning.pm";
+        if (check_screen('ARCH', 'x86_64')) {
+            loadtest "installation/partitioning.pm";
```
Comment (mimi1vx) [inline products/sle/main.pm]: really ? Use system_role only for x86_64 arch

**Comment by dzedro** [comment]
hehe true fail
@mimi1vx what about this ?

**Comment by mimi1vx** [comment]
LGTM

**Comment by okurz** [comment]
I am thinking about proper version comparison methods as part of the test api to do stuff like "if $VERSION > ...". Do we need this?

**Comment by dzedro** [comment]
If version would be just number

**Comment by okurz** [comment]
> If version would be just number

… then we wouldn't need a specific version check

**Comment by coolo** [comment]
we should having helper functions in main.pm at least. these complex boolean constructs tend to grow ;(

**Comment by okurz** [comment]
--> https://progress.opensuse.org/issues/11932

**Comment by coolo** [comment]
why are you merging such things in the middle of Beta validation? This now broke Desktop

**Comment by okurz** [comment]
did not know I should not merge it
1. it should fix issues e.g. on power
2. I thought deployment of changes would be disabled if this is not desired

**Comment by coolo** [comment]
1. there was no problem to be fixed. the test is only made stricter - too strict
2. don't think, ask

**Pull request merged**